src/core/functions/user_functions/UserScriptFunctions.ts

```diff
@@ -66,7 +70,36 @@ export class UserScriptFunctions implements IGenerateObject {
     }
 
     async generate_object(): Promise<Record<string, unknown>> {
-        const user_script_functions = await this.generate_user_script_functions();
+        const user_script_functions =
+            await this.generate_user_script_functions();
         return Object.fromEntries(user_script_functions);
     }
+
+    async inline_requires(file_content: string): Promise<string> {
+        const require = file_content.match(/require\('\.\/(.*)(\.js)?'\)/);
+        if (!require) {
+            return file_content;
+        }
+        if (require && require[1]) {
+            const require_path = `${this.plugin.settings.user_scripts_folder}/${require[1]}.js`;
+            const require_file =
+                this.app.vault.getAbstractFileByPath(require_path);
+            let require_content = await this.app.vault.read(require_file as TFile);
+            require_content = require_content.replace(
+                /module\.exports = .*$/,
+                ""
+            );
+            const exists = file_content.includes(require_content);
+            let inlined = file_content;
+            if (!exists) {
+                inlined = file_content.replace(
+                    `const ${require[1]} = ${require[0]}`,
+                    require_content
+                );
```

Comment:
I just realized you could probably do this:
```suggestion
                inlined = file_content.replace(
                    require[0],
                    require_content
                );
```

Reply:
Well, something *like* that, not exactly that. Maybe if the require regex included the `const` as well.

It makes the regex more complicated (this is certainly not a valid regex for js variable name rules):

```javascript
/const\s+[a-zA-Z0-9\$_]+\s+=\s+require\('\.\/(.*?)(\.js)?'\)/
```